lib/translator/index.js

```diff
@@ -66,6 +63,53 @@ Translator.create = function (acceptLanguage, allTranslations, logger = console,
     return new Translator(acceptLanguage, allTranslations, logger, omitTransforming);
 };
 
+/**
+ * Precompile translation functions
+ * @param {Object} translations
+ * @returns {Object}
+ */
+ Translator.precompileTranslations = function (translations) {
+    const compiled = {};
+
+    Object.keys(translations).forEach(language => {
+        compiled[language] = {}
+        Object.keys(translations[language]).forEach(categoryKey => {
+            compiled[language][categoryKey] = translations[language][categoryKey];
+            if (categoryKey === 'translations') {
+                compiled[language].compiledTranslations = {};
+                Object.keys(translations[language][categoryKey]).forEach(key => {
+                    compiled[language].compiledTranslations[key] = Translator.compileFormatterFunction(translations[language], key);
+                })
+            }
+        });
+    });
+
+    return compiled;
+}
+
+/**
+ * Precompile translation functions
+ * @param {Object} language
+ * @param {String} key
+ * @returns {Object}
+ */
+Translator.compileFormatterFunction = function (language, key) {
+    const locale = language.locales[key];
+    const formatter = new MessageFormat(locale)
+
+    try {
+        return formatter.compile(language.translations[key]).toString();
+    } catch (err) {
+        if (err.name === 'SyntaxError') {
+            this.logger.error(`Syntax Error during Formatter function precompilation: ${err.message} for key "${key}"`, err.expected);
+
+            return () => '';
```

Comment:
Is this not a situation where we want to also throw an error? Why do we just want to return empty string here?

Reply:
I'm not sure why it was there, I've just used the same logic as it was at current helper implementation to have compatibility